tajo-catalog/tajo-catalog-server/src/test/java/org/apache/tajo/catalog/TestCatalogExceptions.java

```diff
@@ -0,0 +1,293 @@
+import org.apache.tajo.util.CommonTestingUtil;
+import org.apache.tajo.util.KeyValueSet;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.net.URI;
+import java.util.UUID;
+
+public class TestCatalogExceptions {
+
+  static CatalogServer server;
+  static CatalogService catalog;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    server = new MiniCatalogServer();
+    catalog = new LocalCatalogWrapper(server);
+    CatalogTestingUtil.prepareBaseData(catalog, ((MiniCatalogServer) server).getTestDir());
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    CatalogTestingUtil.cleanupBaseData(catalog);
+    server.stop();
+  }
+
+  @Test
+  public void testCreateTablespaceWithWrongUri() throws Exception {
+    // TODO
+    catalog.createTablespace("wrong", "hdfs:");
+  }
+
+  @Test(expected = DuplicateTablespaceException.class)
+  public void testCreateDuplicateTablespace() throws Exception {
+    catalog.createTablespace("space1", "hdfs://xxx.com/warehouse");
+  }
+
+  @Test(expected = UndefinedTablespaceException.class)
+  public void testDropUndefinedTablespace() throws Exception {
+    catalog.dropTablespace("undefined");
+  }
+
+  @Test(expected = InsufficientPrivilegeException.class)
+  public void testDropDefaultTablespace() throws Exception {
+    catalog.dropTablespace(TajoConstants.DEFAULT_TABLESPACE_NAME);
+  }
+
+  @Test(expected = TajoInternalError.class)
+  public void testAlterTablespaceWithWrongUri() throws Exception {
+    catalog.alterTablespace(AlterTablespaceProto.newBuilder().
+        setSpaceName("space1").
+        addCommand(
+            AlterTablespaceCommand.newBuilder().
+                setType(AlterTablespaceType.LOCATION).
+                setLocation(SetLocation.newBuilder()
+                    .setUri("hdfs:"))).build());
+  }
+
+  @Test(expected = UndefinedTablespaceException.class)
+  public void testAlterUndefinedTablespace() throws Exception {
+    catalog.alterTablespace(AlterTablespaceProto.newBuilder().
+        setSpaceName("undefined").
+        addCommand(
+            AlterTablespaceCommand.newBuilder().
+                setType(AlterTablespaceType.LOCATION).
+                setLocation(SetLocation.newBuilder()
+                    .setUri("hdfs://zzz.com/warehouse"))).build());
+  }
+
+  @Test(expected = DuplicateDatabaseException.class)
+  public void testCreateDuplicateDatabase() throws Exception {
+    catalog.createDatabase("TestDatabase1", "space1");
+  }
+
+  @Test(expected = UndefinedDatabaseException.class)
+  public void testDropUndefinedDatabase() throws Exception {
+    catalog.dropDatabase("undefined");
+  }
+
+  @Test(expected = InsufficientPrivilegeException.class)
+  public void testDropDefaultDatabase() throws Exception {
+    catalog.dropDatabase(TajoConstants.DEFAULT_DATABASE_NAME);
+  }
+
+  @Test()
+  public void testCreateTableWithWrongUri() throws Exception {
+    // TODO
```

Comment:
It seems to throw some exception, but it does not catch or handle the exception.